xmtp_mls/src/storage/encrypted_store/consent_record.rs

```diff
@@ -57,14 +57,14 @@ impl DbConnection {
         })?)
     }
 
-    /// Insert consent_record, and replace existing entries
-    pub fn insert_or_replace_consent_record(
+    /// Insert consent_records, and replace existing entries
+    pub fn insert_or_replace_consent_records(
         &self,
-        record: StoredConsentRecord,
+        records: Vec<StoredConsentRecord>,
     ) -> Result<(), StorageError> {
         self.raw_query(|conn| {
             diesel::insert_into(dsl::consent_records)
-                .values(&record)
+                .values(&records) // Use a vector of records instead of a single record
```

Comment:
@insipx sqlite doesnt allow this. Would just wrapping this in a transaction and keeping it as before doing it one by one the most performant way?

Reply:
Not sure if below about consent in android is related but wanted to put some context:


this kind of insertion for SQLite database is standard, and the same way diesel works for normal bulk inserts: https://github.com/diesel-rs/diesel/blob/f7e9819303877b5a255143c4b32fa1e5745aaad0/diesel/src/query_builder/insert_statement/insert_with_default_for_sqlite.rs#L241

With WAL logging, this will still result in one overall write to the DB, so would be the best way to get this done I think